src/sentry/utils/integrationdocs.py

```diff
@@ -122,9 +123,14 @@ def sync_docs(quiet=False):
 
     dump_doc("_platforms", {"platforms": platform_list})
 
+    executor = multiprocessing.dummy.Pool(min(32, multiprocessing.cpu_count() * 5))
     for platform_id, platform_data in iteritems(data["platforms"]):
         for integration_id, integration in iteritems(platform_data):
-            sync_integration_docs(platform_id, integration_id, integration["details"], quiet)
+            executor.apply_async(
+                sync_integration_docs, (platform_id, integration_id, integration["details"], quiet)
+            )
+    executor.close()
+    executor.join()
```

Comment:
`multiprocessing.dummy.Pool` doesn't seem to work as a context manager. ThreadPoolExecutor does, and it also makes the code more readable (TIL `multiprocessing.dummy.Pool` uses threading as the "backend", despite its confusing name):

```
    from concurrent.futures import ThreadPoolExecutor

    with ThreadPoolExecutor(max_workers=len(data["platforms"])) as executor:
        for platform_id, platform_data in iteritems(data["platforms"]):
            for integration_id, integration in iteritems(platform_data):
                executor.submit(
                    sync_integration_docs, platform_id, integration_id, integration["details"], quiet
                )
```

I observed no speed difference.

Reply:
We cannot use `ThreadPoolExecutor` because it only became part of stdlib after Python 3.something. We do have the backport of `futures` but we cannot use external dependencies here as this function is called from `setup.py` during wheel building, which is what I'm interested in speeding up.

Reply:
Oh, damn. Could you leave a `TODO(python3)` comment with this explanation?